Improve Message Requests UI for iOS

lourou opened the pull request:
### Changes

- Implemented a segmented controller to switch between "You might know" and "Spam" sections
- Added conditional rendering logic for different scenarios:
  - Both spam and non-spam messages present
  - Only non-spam messages present
  - Only spam messages present
  - No messages present
- Improved UI text to provide context for each scenario

### Screenshots

Attaching dark mode screenshots too:

![Spamonly req button](https://github.com/user-attachments/assets/32da9513-6381-4cdd-851f-46f91e2d9850)

![Simulator Screenshot - iPad Air 11-inch (M2) - 2024-09-03 at 16 10 55](https://github.com/user-attachments/assets/ba27a6a9-a856-4b8b-97e2-bb0013d3bea0)

![IMG_1287](https://github.com/user-attachments/assets/144560f9-3301-48a2-9397-ee05a51d14db)

![IMG_1288](https://github.com/user-attachments/assets/9039917a-c288-4e2b-bf8f-e549655b0fd4)

![Spam Only](https://github.com/user-attachments/assets/b484a50d-4e3f-433c-845c-09949808e762)

Closes [#178](https://github.com/ephemeraHQ/converse-app/issues/178)

darickdang commented:
@lourou Here are some polish recommendations:

![#178 - Requests bar does not match new design](https://github.com/user-attachments/assets/8696aa89-8ac4-4914-aa98-f41df1f081c6)

lourou commented:
Thanks for the feedback @darickdang I've made the changed:
- segmented controller is now smaller in height (32px instead or 40px)
- changed the bold font to semi-bold
- applied the padding as requested
- changed the information text size from 14 to 12
- used a darker gray for the zero request gray button (secondary gray)

lourou commented:
iPhone, main screen:
![Simulator Screenshot - iPhone 15 - 2024-09-05 at 16 40 41](https://github.com/user-attachments/assets/8d6e32b4-3000-41ba-b3a3-61df0373f0a8)

iPhone, with segmented controller:
![Simulator Screenshot - iPhone 15 - 2024-09-05 at 16 40 50](https://github.com/user-attachments/assets/af9d48f2-0b9d-40da-a272-18c408fc33cc)

iPad, without segmented:
![Simulator Screenshot - iPad Air 11-inch (M2) - 2024-09-05 at 16 40 55](https://github.com/user-attachments/assets/55c63c4c-7efd-4d4f-9599-bd6122c4022a)

alexrisch commented on screens/Navigation/ConversationRequestsListNav.ios.tsx:
```diff
@@ -0,0 +1,261 @@
+      navigation,
+    }: {
+      route: RouteProp<NavigationParamList, "ChatsRequests">;
+      navigation: any;
```
Can we avoid anys?

lourou replied:
Changed to `NativeStackNavigationProp<any>;`

alexrisch commented on screens/Navigation/ConversationRequestsListNav.ios.tsx:
```diff
@@ -0,0 +1,261 @@
+    }) => ({
+      animation: navigationAnimation as StackAnimationTypes,
+      headerTitle: clearingAll
+        ? Platform.OS === "ios"
```
This is in a .ios file so no need to check these things

lourou replied:
Thanks, this is fixed!

alexrisch reviewed (approved):
Approving for builds, will make some clean up tickets for comments
